lib/authn.js

```diff
@@ -7,58 +7,28 @@ import {logger} from './logger.js';
 import {MDBE_AUTHZ_FAILED} from './exceptions.js';
 import mongo from 'mongodb';
 import semver from 'semver';
+import {klona} from 'klona';
 
 const {MongoClient} = mongo;
 const {util: {BedrockError}} = bedrock;
 
 export async function openDatabase(options) {
   const config = bedrock.config.mongodb;
+  // copy the config stuff related to connecting
+  const opts = {
+    database: config.name,
+    authentication: {...config.authentication},
+    // authSource should be set in connectOptions
+    connectOptions: {...config.connectOptions},
+    writeOptions: {...config.writeOptions},
+    ...options
+  };
 
-  // connect to database and get server info
-  const {client, db} = await _connect(options);
-  const serverInfo = await db.admin().serverInfo(null);
-
-  // check that server version is supported
-  const {version} = serverInfo;
-  logger.info('connected to database', {
-    url: urls.sanitize(config.url),
-    version,
-  });
-  if(config.requirements.serverVersion) {
-    if(!semver.satisfies(version, config.requirements.serverVersion)) {
-      throw new BedrockError(
-        'Unsupported database version.',
-        'DatabaseError', {
-          url: urls.sanitize(config.url),
-          version,
-          required: config.requirements.serverVersion
-        });
-    }
-  }
-
-  // determine if authN is required
-  let authRequired;
-  if(config.forceAuthentication) {
-    authRequired = true;
-  } else {
-    try {
-      // if listing databases fails on authz, then authentication is required
-      await db.admin().listDatabases(null);
-    } catch(e) {
-      if(e && e.code === MDBE_AUTHZ_FAILED) {
-        authRequired = true;
-      } else {
-        // some other error, abort
-        throw e;
-      }
-    }
-  }
-
-  if(!authRequired) {
-    // return early
-    logger.debug('database authentication not required');
-    return {client, db};
-  }
+  // connects to the mongo server unauthenticated
+  const {admin} = await _getUnauthenticatedDb({config: klona(config)});
+  // gets the serverInfo
```

Comment:
```suggestion
  // do unauthenticated connection to mongo server to check
  // server compatibility and authn requirements
  const {admin} = await _getUnauthenticatedDb({config: klona(config)});
```

Reply:
Yes, the comment suggestion isn't covering only a single line, but the following block of code.